packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1047,10 +1050,16 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
       nuisanceGasLeft = nuisanceGasLeftPostRevert;
     }
 
+    // This check should not be necessary due to the similar check in _useNuisanceGas,
+    // but it's a nice fail-safe to have.
+    if (prevNuisanceGasLeft < nuisanceGasLimit - nuisanceGasLeft) {
+      _revertWithFlag(RevertFlag.EXCEEDS_NUISANCE_GAS);
+    }
+
     // We need to reset the nuisance gas back to its original value minus the amount used here.
     messageRecord.nuisanceGasLeft =
       prevNuisanceGasLeft -
-      (nuisanceGasLimit - nuisanceGasLeft);
+      (nuisanceGasLimit - nuisanceGasLeft); // underflow is here
```

Comment:
I assume after the above changes there would no longer be an underflow

Reply:
Yeah, this is meant to be a temp comment, good call out.